### PR TITLE
Push button, deploy production

### LINKED ISSUE #90: Deployment requires more than a single button press
## Expected behaviour

Deploying changes to the services requires a single button press.

## Actual behaviour

Deploying each service requires a separate list of manual steps.

## Details and resources

Eventually we want to get to CI with Docker. Until then, we need a faster way to ship. This should be as simple as writing a script that can be run on the production EC2 box.

Depends on #51.

## Checklist

- [x] This issue is linked to the appropriate project.
- [x] This issue has appropriate labels.

### PR DESCRIPTION
**Pull request by davidvandusen**
## Proposed changes

This GitHub Actions workflow uses ssh to run the deploy steps on the production server. Namely, run the database migrations and then deploy the docker stack.

Resolves #90, finally.

## Checklist

- [x] Are the issues being addressed linked to this PR?
- [x] Do all commit messages start with the issue number?
- [x] Are all code changes sufficiently tested?
- [x] Are there screenshots for UI changes?

### CODE REVIEW

**Review by BRashad — APPROVED** [review]
I was wondering how these checks are being run in Github. Because of this PR, I learned something new today :). No comments here.

**Review by pthan1 — APPROVED** [review]
Congrats!  Excited to see this workflow up!

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+          Host ${{ github.event.inputs.environment }}
+            HostName $SSH_HOST
+            User $SSH_USER
+            IdentityFile ~/.ssh/${{ github.event.inputs.environment }}.key
+          END
+          ssh-keyscan -H $SSH_HOST >> ~/.ssh/known_hosts
+      - name: Log in to container registry
+        run: |
+          echo "$GITHUB_TOKEN" | ssh ${{ github.event.inputs.environment }} 'docker login ghcr.io --username ${{ github.actor }} --password-stdin'
+      - name: Run database schema migrations
```
[the remark this answers is not in the document]
Comment (davidvandusen) [reply .github/workflows/deploy.yml:42]: The dev environment uses Docker Compose, whereas the prod environment uses Docker Swarm. When creating services as a Docker stack, a minimum number of replicas are ensured, which in the case of flyway would mean every time the migrations finished, they would run again. There used to be a [janky workaround](https://github.com/alexellis/jaas) for this. Docker Compose allows the container to exit and doesn't create a new one.

In the deployment pipeline, if the migrations fail, the new services should not be deployed—there might be a new API endpoint that relies on a table that doesn't exist, or whatever. By using `docker run`, the exit code of the container is available to the workflow runner easily.

In any case, running the migrations as a service would require a more complex deployment configuration on a more capable orchestrator, like Kubernetes. Baby steps.

[the remark this answers is not in the document]
Comment (pthan1) [reply .github/workflows/deploy.yml:42]: Thanks for explaining!

**Pull request merged**